### PR TITLE
Fix linking objc binaries in the root package

### PR DESCRIPTION
**Pull request by keith**
Fixes https://github.com/bazelbuild/bazel/issues/24625

### CODE REVIEW

**Comment by keith** [comment]
@bazel-io flag

File: src/main/starlark/builtins_bzl/common/objc/compilation_support.bzl [hunk 1]
```diff
@@ -28,6 +28,9 @@ load(":common/xcode/providers.bzl", "XcodeVersionInfo")
 
 objc_internal = _builtins.internal.objc_internal
 
+def _join_path(*path):
+    return "/".join([p for p in path if p != ""])
```
Comment (fmeum) [inline src/main/starlark/builtins_bzl/common/objc/compilation_support.bzl:32]: https://github.com/bazelbuild/bazel/blob/48fe8619bf4afbd9d00f2d0ce32b27a950e09767/src/main/starlark/builtins_bzl/common/paths.bzl#L73 is a similar helper that already exists.

Comment (keith) [reply src/main/starlark/builtins_bzl/common/objc/compilation_support.bzl]: thanks!

**Review by fmeum — APPROVED** [review]

**Review by pzembrod — APPROVED** [review]
Thanks a lot, @keith, for chasing this one down!

**Pull request closed without merging**